packages/notifications/src/error.test.ts

```diff
@@ -0,0 +1,121 @@
+import { IgnoreErrorDecider, makeNotificationsContract } from './error';
+import { Notifier } from './useNotifications';
+
+const mockNotifications: Notifier = {
+  notify: jest.fn(),
+};
+
+const ignoreError: IgnoreErrorDecider = (title: string, _obj?: unknown): boolean => {
+  return title === 'Ignore Me';
+};
+describe('reportError', () => {
+  it('calls notify and console.error', async () => {
+    // Arrange
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+    const { reportError } = makeNotificationsContract(mockNotifications, ignoreError);
+
+    // Act
+    await reportError('Things went BOOM', new Error('BOOM!'));
+
+    // Assert
+    const expectedError = new Error('BOOM!');
+    expect(mockNotifications.notify).toBeCalledTimes(1);
+    expect(mockNotifications.notify).toBeCalledWith('error', 'Things went BOOM', { detail: expectedError });
+    expect(console.error).toBeCalledTimes(1);
+    expect(console.error).toBeCalledWith('Things went BOOM', expectedError);
+  });
+
+  it('ignores session timeout error', async () => {
+    // Arrange
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+    const { reportError } = makeNotificationsContract(mockNotifications, ignoreError);
+
+    // Act
+    await reportError('Ignore Me', new Error('I cried wolf.'));
+
+    // Assert
+    expect(mockNotifications.notify).toBeCalledTimes(0);
+    expect(console.error).toBeCalledTimes(1);
+    expect(console.error).toBeCalledWith('Ignore Me', new Error('I cried wolf.'));
+  });
+});
+
+describe('reportErrorAndRethrow', () => {
+  it('calls notify and rethrows error', async () => {
+    // Arrange
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+    const { reportErrorAndRethrow } = makeNotificationsContract(mockNotifications, ignoreError);
+    const callMe = reportErrorAndRethrow('Things went BOOM')(async () => {
+      throw new Error('BOOM!');
+    });
+    let outerThrow: unknown | null = null;
+
+    // Act
+    try {
+      await callMe();
+    } catch (e) {
+      outerThrow = e;
+    }
```

Comment:
Consider using [toThrow](https://jestjs.io/docs/expect#tothrowerror) to check for an expected exception.

```suggestion
    await expect(callMe).rejects.toThrow(expectedError);
```

Reply:
I go back and forth on this... I think the current form reads better through the AAA sections... whereas using expect....toThrow technically has the Act in the Assert section....

Reply:
🤷 I just write the comment as "Act / Assert". It's a lot more concise and avoids the mutable variable.

Reply:
ok yeah, I'll switch them over.